File: api/database/enum/difficulty.go
package enum

import (
	"database/sql/driver"
	"fmt"
)

type Difficulty string

const (
	Trivial        Difficulty = "trivial"
	Easy           Difficulty = "easy"
	Medium         Difficulty = "medium"
	Challenging    Difficulty = "challenging"
	Hard           Difficulty = "hard"
	Extreme        Difficulty = "extreme"
	SuicideMission Difficulty = "suicide_mission"
	Impossible     Difficulty = "impossible"
	Helldive       Difficulty = "helldive"
)

func (d *Difficulty) Scan(value interface{}) error {
	b, ok := value.([]byte)
	if !ok {
		switch value {
		case Trivial, Easy, Medium, Challenging, Hard, Extreme, SuicideMission, Impossible, Helldive:
			*d = Difficulty(b)
		default:
			return fmt.Errorf("invalid value for Difficulty: %v", value)
		}
	}
	return nil
}

func (d Difficulty) Value() (driver.Value, error) {
	return string(d), nil
}

File: web/lib/enum/difficulty.dart
import 'package:flutter/material.dart';
import 'package:flutter_gen/gen_l10n/app_localizations.dart';
import 'package:json_annotation/json_annotation.dart';

enum Difficulty {
  @JsonValue('trivial')
  trivial(
    'Trivial',
  ),

  @JsonValue('easy')
  easy(
    'Easy',
  ),

  @JsonValue('medium')
  medium(
    'Medium',
  ),

  @JsonValue('challenging')
  challenging(
    'Challenging',
  ),

  @JsonValue('hard')
  hard(
    'Hard',
  ),

  @JsonValue('extreme')
  extreme(
    'Extreme',
  ),

  @JsonValue('suicide_mission')
  suicideMission(
    'Suicide Mission',
  ),

  @JsonValue('impossible')
  impossible(
    'Impossible',
  ),

  @JsonValue('helldive')
  helldive(
    'Helldive',
  );

  const Difficulty(
    this.name,
  );

  /// The untranslated name of the difficulty. Use [translatedName] to get the translated name.
  final String name;

  String translatedName(BuildContext context) {
    switch (this) {
      case Difficulty.easy:
        return AppLocalizations.of(context)!.difficultyTrivial;
      case Difficulty.trivial:
        return AppLocalizations.of(context)!.difficultyEasy;
      case Difficulty.medium:
        return AppLocalizations.of(context)!.difficultyMedium;
      case Difficulty.challenging:
        return AppLocalizations.of(context)!.difficultyChallenging;
      case Difficulty.hard:
        return AppLocalizations.of(context)!.difficultyHard;
      case Difficulty.extreme:
        return AppLocalizations.of(context)!.difficultyExtreme;
      case Difficulty.suicideMission:
        return AppLocalizations.of(context)!.difficultySuicideMission;
      case Difficulty.impossible:
        return AppLocalizations.of(context)!.difficultyImpossible;
      case Difficulty.helldive:
        return AppLocalizations.of(context)!.difficultyHelldive;
    }
  }
}
